### PR TITLE
Upgrade Table authoring to Angular component

### LINKED ISSUE #2902: Upgrade Table authoring to Angular component
Upgrade the AngularJS component to Angular.

### PR DESCRIPTION
**Pull request by geoffreykwan**
Make sure Table authoring still works.

Closes #2902

### CODE REVIEW

File: src/main/webapp/wise5/components/table/table-authoring/table-authoring.component.ts [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+  insertColumn(x: number): void {
+    const tableData = this.authoringComponentContent.tableData;
+    const numRows = this.authoringComponentContent.numRows;
+    for (let r = 0; r < numRows; r++) {
+      const tempRow = tableData[r];
+      if (tempRow != null) {
+        const newCell = this.createEmptyCell();
+        tempRow.splice(x, 0, newCell);
+      }
+    }
```
Comment (hirokiterashima) [inline src/main/webapp/wise5/components/table/table-authoring/table-authoring.component.ts:249]: Is this null check necessary?

File: src/main/webapp/wise5/components/table/table-authoring/table-authoring.component.ts [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+  deleteColumn(x: number): void {
+    if (confirm($localize`Are you sure you want to delete this column?`)) {
+      const tableData = this.authoringComponentContent.tableData;
+      const numRows = this.authoringComponentContent.numRows;
+      for (let r = 0; r < numRows; r++) {
+        const tempRow = tableData[r];
+        if (tempRow != null) {
+          tempRow.splice(x, 1);
+        }
+      }
```
Comment (hirokiterashima) [inline src/main/webapp/wise5/components/table/table-authoring/table-authoring.component.ts:264]: Is this null check necessary?

**Review by hirokiterashima — CHANGES_REQUESTED** [review]
Functionality looks good. I cleaned up the code a little bit. Also found a part of the code where I wasn't sure if the null-check is necessary and added comments inline. Can you take a look?

**Comment by geoffreykwan** [comment]
@hirokiterashima Ok I removed those null checks because they didn't seem to be needed. I also set the min value for inputs to 1 and changed some variable names.

**Review by hirokiterashima — APPROVED** [review]
Looks good.

**Pull request merged**